### DISCUSSION TITLE
Bézier curve not taking width parameter

### DISCUSSION THREAD

**Discussion by pl-deassis** [gdsfactory/gdsfactory#2323] [Q&A] [answered]
I am having an issue where in one PC I can get a script to run where a Bézier curve accepts a width parameter for the waveguide, but when the same code is run in another PC, it complains and will not run.

The same problem is manifesting when I try to pass a radius argument to a delay snake. It works fine in one PC but not in another (with a fresh installation of gdsfactory).

**Comment by joamatab** [comment]
Are you sure you are not using different versions of gdsfactory on each computer?

**Reply by pl-deassis** [reply]
I just found out (after about 3 hours of looking for other causes) that this is the case! The code works on v. 6.104.2... In the new version, several components, such as s-bends and delay lines, no longer accept the kwargs I was passing onto them. I saw on the changelog that unused kwargs were removed, but how can I replace this functionality?

I tried to explicitly define a cross section, but this lead to even more work, because ports that were automatically placed on components are no longer there if I don't list the ports on the sections. I tried to read up on how the cross sections work in order to recover this functionality, but with little success.

**Reply by pl-deassis** [reply]
Oddly enough, the delay_snake2 component still accepts a width argument, but not the radius argument, which I find to be quite useful...

**Comment by joamatab** [comment, marked as answer]
passing kwargs to modify component cross_section was a convenient hack that we have been deprecating 

the clean way is to define new cross_sections with the correct arguments


```python
from functools import partial
import gdsfactory as gf

xs_sc1 = partial(gf.cross_section.strip, width=1)
xs_sc2 = partial(gf.cross_section.strip, width=2)
xs_sc_clad = partial(
    gf.cross_section.strip, cladding_layers=[(3, 0)], cladding_offsets=[3]
)

c = gf.grid(
    [gf.components.straight(cross_section=xs) for xs in [xs_sc1, xs_sc2, xs_sc_clad]]
)
c.show()
```

![image](https://github.com/gdsfactory/gdsfactory/assets/4514346/6c5fdfc5-b159-4398-b75a-6ffdc57895be)
